ride-share.rb

```diff
@@ -0,0 +1,144 @@
+RIDE_SHARE = {
+    'DR0004': [
+        {
+            date: '3rd Feb 2016',
+            cost: 5,
+            rider_id: 'RD0022',
+            rating: 5
+        },
+        {
+            date: '4th Feb 2016',
+            cost: 10,
+            rider_id: 'RD0022',
+            rating: 4
+        },
+        {
+            date: '5th Feb 2016',
+            cost: 20,
+            rider_id: 'RD0073',
+            rating: 5   
+        }
+    ],
+    'DR0001': [
+        {
+            date: '3rd Feb 2016',
+            cost: 10,
+            rider_id: 'RD0003',
+            rating: 3            
+        },
+        {
+            date: '3rd Feb 2016',
+            cost: 30,
+            rider_id: 'RD0015',
+            rating: 4
+        },
+        {
+            date: '5th Feb 2016',
+            cost: 45,
+            rider_id: 'RD0003',
+            rating: 2
+        }
+    ],
+    'DR0002': [
+        {
+            date: '3rd Feb 2016',
+            cost: 25,
+            rider_id: 'RD0073',
+            rating: 5            
+        },
+        {
+            date: '4th Feb 2016',
+            cost: 15,
+            rider_id: 'RD0013',
+            rating: 1
+        },
+        {
+            date: '5th Feb 2016',
+            cost: 35,
+            rider_id: 'RD0066',
+            rating: 3
+        }
+    ],
+    'DR0003': [
+        {
+            date: '4th Feb 2016',
+            cost: 5,
+            rider_id: 'RD0066',
+            rating: 5            
+        },
+        {
+            date: '5th Feb 2016',
+            cost: 50,
+            rider_id: 'RD0003',
+            rating: 2
+        }
+    ]
+}
+
+def ride_share_summary(ride_share)
+    puts "Summary: "
+
+    average_rating_list = {}    # key is averate rate and value is driver id
+    money_earned_list = {}  # key is income and value is driver id
+
+    ride_share.each do |driver, rides|
+        this_driver_earned_total = 0
+        this_driver_rating_total = 0
+        this_driver_daily_income = {} # key is date, value is income on this date
```

Comment:
I like how specific your variable names are. There's almost too much information here - if you were to drop the `this_driver_`, the code might be a little more readable. Because the variable is scoped to the loop, the reader can figure out that the `earned_total` is specific to this driver.